osu.Game/Users/UserActivity.cs

```diff
@@ -90,9 +98,34 @@ public Editing(IBeatmapInfo info)
             public override string Status => @"Editing a beatmap";
         }
 
-        public class Spectating : UserActivity
+        public class Watching : UserActivity
         {
-            public override string Status => @"Spectating a game";
+            private readonly ScoreInfo score;
+
+            private string username => score.User.Username;
+            private string playingVerb => score.BeatmapInfo.Ruleset.CreateInstance().PlayingVerb;
+
+            public BeatmapInfo BeatmapInfo => score.BeatmapInfo;
+
+            public Watching(ScoreInfo score)
+            {
+                this.score = score;
+            }
+
+            protected virtual string Verb => @"Watching";
+
+            public override string Status => @$"{Verb} {username} {playingVerb.ToLowerInvariant()}";
```

Comment:
I hope nobody ever decides to localise this, because if they do, then good luck with this one.

Not to mention that it reads wrong to me ("Watching someone mashing keys" - the double infinitive is messing with me, I'd expect "Watching someone mash keys") - but it *does* seem to be valid English.

Also this uses `@$`, then the next line does `$@`. I'd pick one order and stick with it. I think we mostly do the second.

Reply:
Should I keep it simple and just put `{Verb} {username}`?